simconvoi.cc

```diff
@@ -5330,8 +5336,8 @@ bool convoi_t::can_start_coupling(convoi_t* parent) const {
 		// all schedule entries are waypoint.
 		return false;
 	}
-	const schedule_entry_t t_c = schedule->at(t_idx);
-	const schedule_entry_t t_n = schedule->at((t_idx+1)%schedule->get_count());
+	const schedule_entry_t t_c = (schedule->check_next_line_valid()&&t_idx==schedule->get_count()-1)?schedule->get_next_line()->get_schedule()->at(0):schedule->at(t_idx);
+	const schedule_entry_t t_n = (schedule->check_next_line_valid()&&t_idx==schedule->get_count()-1)?schedule->get_next_line()->get_schedule()->at(1):schedule->at((t_idx+1)%schedule->get_count());
```

Comment:
(nit)条件を分離してみやすくした方がよさそうです。
```suggestion
	const bool is_at_dummy_entry = schedule->check_next_line_valid()&&t_idx==schedule->get_count()-1;
	const schedule_entry_t t_c = is_at_dummy_entry ? schedule->get_next_line()->get_schedule()->at(0) : schedule->at(t_idx);
	const schedule_entry_t t_n = is_at_dummy_entry ? schedule->get_next_line()->get_schedule()->at(1) : schedule->at((t_idx+1)%schedule->get_count());
```